### ISSUE TITLE
Chinese Pretraining

### ISSUE THREAD

**Issue by Sara-HY** [google-research/bert#259]
Hi, I am pretraining Chinese bert model. But the loss of the model cannot drop down when it get 7.  And I am not sure what the key problem is.

**Comment by guoyaohua** [comment]
兄弟，请教一下，bert-base 可以跑在两个8g 显存的gpu上么？作者说只能显存大于12G的单GPU，但是使用两个GPU，可以达到扩充显存的作用么？即使只用一个GPU做运算

**Comment by breakjiang** [comment]
> Hi, I am pretraining Chinese bert model. But the loss of the model cannot drop down when it get 7. And I am not sure what the key problem is.

more num_train_steps

**Comment by Sara-HY** [comment]
> > Hi, I am pretraining Chinese bert model. But the loss of the model cannot drop down when it get 7. And I am not sure what the key problem is.
> 
> more num_train_steps

How many num_train_steps have you tried?

**Comment by jockeyyan** [comment]
Depends on how many GPUs you have, as google says: "Pre-training is fairly expensive (four days on 4 to 16 Cloud TPUs)"

**Comment by breakjiang** [comment]
> > > Hi, I am pretraining Chinese bert model. But the loss of the model cannot drop down when it get 7. And I am not sure what the key problem is.
> > 
> > 
> > more num_train_steps
> 
> How many num_train_steps have you tried?

500000

**Comment by Sara-HY** [comment]
> > > > Hi, I am pretraining Chinese bert model. But the loss of the model cannot drop down when it get 7. And I am not sure what the key problem is.
> > > 
> > > 
> > > more num_train_steps
> > 
> > 
> > How many num_train_steps have you tried?
> 
> 500000

Could you please tell me your parameters for pre-trainining? And also, how many instances have you created for pre-training and how may GPUs do you use while training?

**Comment by Sara-HY** [comment]
> Depends on how many GPUs you have, as google says: "Pre-training is fairly expensive (four days on 4 to 16 Cloud TPUs)"

What you said means it is time-consuming. But what I want to know is how many epochs I should train the model so that the loss can drop down.

**Comment by Sara-HY** [comment]
> > > > > > Hi, I am pretraining Chinese bert model. But the loss of the model cannot drop down when it get 7. And I am not sure what the key problem is.
> > > > > 
> > > > > 
> > > > > more num_train_steps
> > > > 
> > > > 
> > > > How many num_train_steps have you tried?
> > > 
> > > 
> > > 500000
> > 
> > 
> > Could you please tell me your parameters for pre-trainining? And also, how many instances have you created for pre-training and how may GPUs do you use while training?
> 
> 1 million instances
> 1 GPU has 64GB of RAM
> 0.26 million vocab size
> model parameter L-4-H-288-A-12
> batch size 32
> 
> If you have larger GPU RAM ，you can set larger hidden_size

Thank you so much for your reply.  Maybe I have something wrong on the pre-processing on the raw data for creating the instances. How about this part? Do you use wiki data directly? What else preprocessing have you done?

**Comment by qiunian711** [comment]
> > > > > > Hi, I am pretraining Chinese bert model. But the loss of the model cannot drop down when it get 7. And I am not sure what the key problem is.
> > > > > 
> > > > > 
> > > > > more num_train_steps
> > > > 
> > > > 
> > > > How many num_train_steps have you tried?
> > > 
> > > 
> > > 500000
> > 
> > 
> > Could you please tell me your parameters for pre-trainining? And also, how many instances have you created for pre-training and how may GPUs do you use while training?
> 
> 1 million instances
> 1 GPU has 8GB of RAM
> 0.26 million vocab size
> model parameter L-4-H-288-A-12
> batch size 32
> 
> If you have larger GPU RAM ，you can set larger hidden_size

@breakjiang 
Hello, I am also processing Chinese corpus. Is your vocab.txt file generated during data processing based on your own corpus?
Are you training on the basis of the pre-trained Chinese model provided by Google? The size of the vocab in the pre-trained model is 21122. This seems to be fixed. I don't know how you handled it.
Is 4 in "model parameter L-4-H-288-A-12" indicating 4 transformer layers, and 288 is the number of hidden units. Is this understood?

**Comment by stevewyl** [comment]
@qiunian711 Why to use own word dictionary？Bert-Chinese is character-based. Use the word
dictionary may largely increase model size. I just continue pre-train bert model on my own corpus for only one epoch. And the f1 score of fine-tuned model got 1% improvement.

**Comment by qiunian711** [comment]
@stevewyl   Because my ultimate goal is to extract the vector of words like word2vec instead of the vector of characters, I want to train directly based on my own dictionary. This is mainly due to the fact that if a word is expressed in a few characters in Chinese, the meaning may be greatly deviated, so I dare not train directly based on characters. Do you have any good suggestions for this?

**Comment by biuleung** [comment]
Have you met these errors while you fine tuned the self pre-trained model?

2 root error(s) found.
  (0) Not found: Key output_bias not found in checkpoint
	 [[node save/RestoreV2 (defined at run_classifier.py:953) ]]
	 [[save/RestoreV2/_301]]
  (1) Not found: Key output_bias not found in checkpoint
	 [[node save/RestoreV2 (defined at run_classifier.py:953) ]]
0 successful operations.
0 derived errors ignored.

**Comment by khaerulumam42** [comment]
Anyone who get access on TPU for free, I granted access from TFRC programs and get 100 TPU v2 for free. I have still many TPU to use, maybe you guys want it, I hope it will helps you. My access deadline just up to 24 July.

**Comment by FengXuas** [comment]
对中文预训练，需要对中文语料分词吗？
